appservice/src/SiteWrapper.ts

```diff
@@ -170,6 +176,61 @@ export class SiteWrapper {
         outputChannel.appendLine('');
     }
 
+    public async connectToGitHub(node: IAzureNode, outputChannel: vscode.OutputChannel): Promise<void> {
+
+        type gitHubOrgData = { repos_url?: string };
+        type gitHubReposData = { repos_url?: string, url?: string, html_url?: string };
+
+        const client: WebSiteManagementClient = nodeUtils.getWebSiteClient(node);
+        const oAuth2Token: string = (await client.listSourceControls())[0].token;
+        if (!oAuth2Token) {
+            this.showGitHubAuthPrompt(node);
+            return;
+        }
+
+        const gitHubUser: Object[] = await this.getJsonRequest('https://api.github.com/user', oAuth2Token, node);
+
+        const gitHubOrgs: Object[] = await this.getJsonRequest('https://api.github.com/user/orgs', oAuth2Token, node);
+        const orgQuickPicks: IQuickPickItemWithData<{}>[] = this.createQuickPickFromJsons([gitHubUser], 'login', undefined, ['repos_url']).concat(this.createQuickPickFromJsons(gitHubOrgs, 'login', undefined, ['repos_url']));
+        const orgQuickPick: gitHubOrgData = (await vscode.window.showQuickPick(orgQuickPicks, { placeHolder: 'Choose your organization.', ignoreFocusOut: true })).data;
```

Comment:
Feels like you're not handling the case if the user cancels. Aka it should throw a UserCancelledError